### PR TITLE
Feature: Add support for referencing class method as callback function

### PR DESCRIPTION
**Pull request by yash-goli**
can pass the name of the member function of class to the onChange in string.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,22 @@ class PersonComponent {
...
+### Example 4, using class method reference for onChange
+```typescript
+class PersonComponent {
+ 
+    @OnChange<string>('onNameChange')  
+    name: string;
+    
+    age: number;
+
+    onNameChange(this: PersonComponent, value, change) {
```
Comment (zhaosiyang) [inline README.md]: We don't need type `this` anymore, so we can remove `this: PersonComponent`

File: src/index.ts [hunk 2]
```diff
@@ -5,18 +5,21 @@ export interface SimpleChange<T> {
...
     const cachedValueKey = Symbol();
     const isFirstChangeKey = Symbol();
+
+    const isFunction = (fn: any) => fn.constructor === Function;
+
     return (target: any, key: PropertyKey) => {
         Object.defineProperty(target, key, {
             set: function (value) {
+
+                const callBackFn = (isFunction(callback) ? callback : target[callback as string]) as Function;
```
Comment (zhaosiyang) [inline src/index.ts]: Please throw an error when the function doesn't exist.

**Review by zhaosiyang — CHANGES_REQUESTED** [review]
Nice solution. I really like this PR. I added some change requests, would you please fix them and I am gonna merge it. Thanks.

File: README.md (hunk 1 above)
Comment (yash-goli) [reply README.md]: Removed the this parameter

File: src/index.ts (hunk 2 above)
Comment (yash-goli) [reply src/index.ts]: Added the exception handling when developer passes invalid string to OnChange decorator

**Comment by yash-goli** [comment]
Changes done as per comments

**Review by zhaosiyang — APPROVED** [review]
Thanks for contributing, I will make some minor updates soon. Merging this PR.

**Pull request merged**